src/cmd/cloud.js

```diff
@@ -41,311 +43,381 @@ const PLATFORMS = extend(utilities.knownPlatforms(), {
-				const fileMapping = this._handleMultiFileArgs(files, { followSymlinks });
-				api._populateFileMapping(fileMapping);
-				return fileMapping;
-			}).then((fileMapping) => {
-				if (Object.keys(fileMapping.map).length === 0) {
-					throw new Error('no files included');
-				}
+							for (let i = 0, n = list.length; i < n; i++){
+								this.ui.stdout.write(`    ${list[i]}${os.EOL}`);
+							}
 
-				if (settings.showIncludedSourceFiles) {
-					const list = _.values(fileMapping.map);
-					console.log('Including:');
-					for (let i = 0, n = list.length; i < n; i++) {
-						console.log('    ' + list[i]);
-					}
-				}
+							this.ui.stdout.write(os.EOL);
+						}
 
-				return this._doFlash({ api, deviceId, fileMapping, targetVersion });
+						return this._doFlash({ product, deviceId: device, fileMapping, targetVersion });
+					});
+			})
+			.catch((error) => {
+				const message = `Failed to flash ${device}`;
+				throw createAPIErrorResult({ error, message });
 			});
-		}).catch((err) => {
-			throw new VError(ensureError(err), 'Flash device failed');
-		});
 	}
 
-	_doFlash({ api, deviceId, fileMapping, targetVersion }) {
-		return Promise.resolve().then(() => {
-			return api.flashDevice(deviceId, fileMapping, targetVersion);
-		}).then((resp) => {
-			if (resp.status || resp.message) {
-				console.log('Flash device OK: ', resp.status || resp.message);
-			} else if (resp.output === 'Compiler timed out or encountered an error') {
-				console.log('\n' + (resp.errors && resp.errors[0]));
-				throw new Error('Compiler encountered an error');
-			} else {
-				throw api.normalizedApiError(resp);
-			}
-		}).catch(err => {
-			throw api.normalizedApiError(err);
-		});
+	_doFlash({ product, deviceId, fileMapping, targetVersion }){
+		return Promise.resolve()
+			.then(() => {
+				if (!product){
+					return;
+				}
+				this.ui.stdout.write(`marking device ${deviceId} as a development device${os.EOL}`);
+				return createAPI().markAsDevelopmentDevice(deviceId, true, product);
+			})
+			.then(() => {
+				this.ui.stdout.write(`attempting to flash firmware to your device ${deviceId}${os.EOL}`);
+				return createAPI().flashDevice(deviceId, fileMapping, targetVersion, product);
+			})
+			.then((resp) => {
+				if (resp.status || resp.message){
+					this.ui.stdout.write(`Flash device OK: ${resp.status || resp.message}${os.EOL}`);
+				} else if (resp.output === 'Compiler timed out or encountered an error'){
+					this.ui.stdout.write(`${os.EOL}${(resp.errors && resp.errors[0])}${os.EOL}`);
+					throw new Error('Compiler encountered an error');
+				} else {
+					throw normalizedApiError(resp);
+				}
+			})
+			.then(() => {
+				if (!product){
+					return;
+				}
+				[
+					`device ${deviceId} is now marked as a developement device and will NOT receive automatic product firmware updates.`,
+					'to resume normal updates, please visit:',
+					`https://console.particle.io/${product}/devices/unmark-development/${deviceId}`
```

Comment:
You'll update this message when you add the command to set/clear a device for development, right?

Reply:
yup - added a `TODO` in https://github.com/particle-iot/particle-cli/pull/565/commits/934d2e14bf3ce2ee5cf1d295117107fc939be7e0 👍